LibreNMS/Device/YamlDiscovery.php

```diff
@@ -127,6 +118,51 @@ public static function discover(OS $os, $class, $yaml_data)
         return $items;
     }
 
+    /**
+     * @param array $device Device we are working on
+     * @param array $data Array derived from YAML
+     * @return string
+     */
+    public static function computeNumericalOID($device, $data)
+    {
+        static $num_oid_cache;
```

Comment:
Could use the actual cache, to cache things across runtimes.

Reply:
Took me a bit of time to find out how, but I think I got it. Caching is done in `static::oidToNumeric` so all calls to this function get cached. And the resulting code is cleaner. 
For the caching time, I went for 30 minutes. Discovery process can take a long time, and mibs are not changing very often really. We could even push this caching time to a few hours I suppose. 

Let me know what you think.